src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -94,44 +97,62 @@ sealed class InteractiveTxInput {
+    data class LocalLegacySwapIn(override val serialId: Long, override val previousTx: Transaction, override val previousTxOutput: Long, override val sequence: UInt, val userKey: PublicKey, val serverKey: PublicKey, val refundDelay: Int) : Local() {
         override val outPoint: OutPoint = OutPoint(previousTx, previousTxOutput)
         override val txOut: TxOut = previousTx.txOut[previousTxOutput.toInt()]
     }
 
+    data class LocalSwapIn(
+        override val serialId: Long,
+        override val previousTx: Transaction,
+        override val previousTxOutput: Long,
+        override val sequence: UInt,
+        val swapInParams: TxAddInputTlv.SwapInParams
+    ) : Local() {
+        override val outPoint: OutPoint = OutPoint(previousTx, previousTxOutput)
+    }
+
     /**
      * A remote input that funds the interactive transaction.
      * We only keep the data we need from our peer's TxAddInput to avoid storing potentially large messages in our DB.
      */
-    sealed class Remote : InteractiveTxInput(), Incoming {
-        abstract val txOut: TxOut
-    }
+    sealed class Remote : InteractiveTxInput(), Incoming
 
     /** A remote-only input that funds the interactive transaction. */
     data class RemoteOnly(override val serialId: Long, override val outPoint: OutPoint, override val txOut: TxOut, override val sequence: UInt) : Remote()
 
     /** A remote input from a swap-in: our peer needs our signature to build a witness for that input. */
-    data class RemoteSwapIn(override val serialId: Long, override val outPoint: OutPoint, override val txOut: TxOut, override val sequence: UInt, val userKey: PublicKey, val serverKey: PublicKey, val refundDelay: Int) : Remote()
+    data class RemoteLegacySwapIn(override val serialId: Long, override val outPoint: OutPoint, override val txOut: TxOut, override val sequence: UInt, val userKey: PublicKey, val serverKey: PublicKey, val refundDelay: Int) : Remote()
+
+    data class RemoteSwapIn(
+        override val serialId: Long,
+        override val outPoint: OutPoint,
+        override val txOut: TxOut,
+        override val sequence: UInt,
+        val swapInParams: TxAddInputTlv.SwapInParams
+    ) : Remote()
 
     /** The shared input can be added by us or by our peer, depending on who initiated the protocol. */
-    data class Shared(override val serialId: Long, override val outPoint: OutPoint, override val sequence: UInt, val localAmount: MilliSatoshi, val remoteAmount: MilliSatoshi) : InteractiveTxInput(), Incoming, Outgoing
+    data class Shared(override val serialId: Long, override val outPoint: OutPoint, override val txOut: TxOut, override val sequence: UInt, val localAmount: MilliSatoshi, val remoteAmount: MilliSatoshi) : InteractiveTxInput(), Incoming,
```

Comment:
We don't need to add the whole `txOut` field, just its `publicKeyScript`, like we did for `eclair`.

Reply:
done in https://github.com/ACINQ/lightning-kmp/pull/563/commits/4237120872025646386ada61fe5d2f7f9d359538